### ISSUE TITLE
The input parameters do not get passed to the Appsync query

### ISSUE THREAD

**Issue by r-supriya** [aws-amplify/aws-sdk-android#1269]
** Which Category is your question related to? **
Amplify- AppSync Query

** What AWS Services are you utilizing? **
Amplify, AppSync, DynamoDB, Cognito

I'm trying to use AppSync to query a DynamoDB database from my Android app. I'm following the steps as in: https://aws-amplify.github.io/docs/android/api#run-a-query
After running this code:
```
mAWSAppSyncClient().query(GetBooksQuery.builder().pK("BOOKc5585804@urhen.com").build())
                .responseFetcher(AppSyncResponseFetchers.NETWORK_ONLY)
                .enqueue(queryCallback); 
```
I expect the response in the GraphQL Callback to be in the form:
```
{
  "data": {
    "getBooks": {
      "items": [
        {
          "PK" : ...
          "SK":...
        }]}}}
```
so that i can access the items using- `response.data().getBooks().items()` 
But in my app, `response.toString()` gives `com.apollographql.apollo.api.Response@4bbebf0`  
and the `response.data()` is always **null** for all queries. 

However while running the same query in the AppSync console, I get the expected response with data and items.
Why is the `response.data()` always null for the queries from the android app? And how do i get the required response?
(I am a beginner. Any help would be appreciated. Thanks)

Edit: Based on @kvasukib 's comment, I could understand from` response.hasErrors()` that the value of PK is not passed on to the query. That's why I get null as the response data. All values that I pass in the query's builder statement are ignored. Why is that so? 

(The workaround that I currently follow is to use a GetBooksQueryInput type, build the input separately and then pass it to the query builder as -
```
GetBooksQueryInput getBooksQueryInput =GetBooksQueryInput.builder().pK("BOOKc5585804@urhen.com").build();
 ClientFactory.appSyncClient().query(GetBooksQuery.builder().input(getBooksQueryInput.build()).responseFetcher(AppSyncResponseFetchers.NETWORK_ONLY)
                .enqueue(queryCallback);
```
This works fine for queries, but becomes a problem with Subscriptions, because subscriptions do not work with input types even in the console. i.e. the subscription `onAddBook(PK: String!): Book` works but `onAddBook(input : onAddBookInput!): Book` with `input onAddBookInput { PK: String! }` doesn't work. )

**Comment by kvasukib** [comment]
@r-supriya Thank you for reporting to us. 

1) Have you tried with other ResponseFetchers such as NETWORK_ONLY, NETWORK_FIRST, CACHE_ONLY? Can you observe the behavior with other response fetchers?

2) What Authorization Mode do you use? Are you using the same Authorization Mode on both the AppSync console as well as the SDK?

3) Do you see any errors in the response when data() is null? Can you check `response.hasErrors()`?

**Comment by r-supriya** [comment]
@kvasukib 
1. Yes, I have tried with other ResponseFetchers. They all give a similar response -with null data.
2. I use AMAZON_COGNITO_USER_POOLS on both, the console as well as the SDK.
3. Yes, The response.hasErrors() is true. The error message is:
`[Error{message='Variable 'PK' has coerced Null value for NonNull type 'String!'', locations=[Location{line=1, column=16}], customAttributes={}}]`
 But I specifically mention the value of PK while building the query as follows: 
```
mAWSAppSyncClient().query(GetBooksQuery.builder().pK("BOOKc5585804@urhen.com").build())
                .responseFetcher(AppSyncResponseFetchers.NETWORK_ONLY)
                .enqueue(queryCallback);
```
Note:
I am trying to store users and books in the same DynamoDB table as 'Entity's with different PK's and SK's. (only one @ model- 'Entity' in the schema, but multiple types- User, Book, etc) I manually created some entries in the DynamoDB table so that I can query. The GetBooks query is supposed to get all the books corresponding to a particular user. (i.e all the'Entity's in the table that have the PK -  "BOOKc5585804@urhen.com"). I have attached the schema, Query.getBooks.req.vtl and Query.getBooks.res.vtl that I use.
[schema and resolvers.zip](https://github.com/aws-amplify/aws-sdk-android/files/3766129/schema.and.resolvers.zip)

My workflow for reference:
1. I created the app and installed dependencies. I installed CLI at the root directory, ran `amplify init` and `amplify add api` and followed the steps. I typed a basic schema in the backend api folder and ran `amplify push`. (with - Continue? Yes; Update code?  Yes;   Generate GraphQL statements? Yes)
2. Next, I tested in the console by adding more attributes, types, and resolvers.
3. Once they worked, I made the same changes in the Amplify backend folder's schema; Pasted the required resolvers in the backend - resolvers folder; and ran `amplify push` again. (Continue? Yes; Update Code?  Yes;   Generate and overwrite GraphQL statements?  No; //This is because I didn't want any existing resolvers and mutations to be overwritten)
4. Then I ran `amplify codegen` in the root folder of my app and build my app. //This is because even though the changes were seen in the backend after the push, I couldn't find the newly added types in the app unless I ran codegen. i.e The file -(MyApp\app\build\generated\source\appsync\com\amazonaws\amplify\generated\graphql\GetBooksQuery.java) was available only after codegen.
I am not sure if this is the right workflow.

**Comment by Nicolas1006** [comment]
Hi, i had the same problem and i could resolve following this guides:
- https://aws.amazon.com/es/blogs/mobile/building-an-android-app-with-aws-amplify-part-1/
- https://docs.amplify.aws/lib/auth/working-with-api/q/platform/android

i understand that you need a list of your data... In this case you need write a code like:

String nexToken = null;

mAWSAppSyncClient().query(ListBooksQuery.builder()
.limit(10000)            .filter(ModelBooksFilterInput.builder().pK(ModelStringInput.builder().eq("BOOKc5585804@urhen.com").build()).build())
.nextToken(nextToken)
                .responseFetcher(AppSyncResponseFetchers.CACHE_AND_NETWORK)
                .enqueue(queryCallback);


the filter is optional, you can comment this line and give all data from toy table.

i hope it's works

**Comment by richardmcclellan** [comment]
Closing due to inactivity.  Please [open a new issue in the aws-mobile-appsync-sdk-android repo](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/new/choose) if you are still having an issue.
